flycheck.el

```diff
@@ -6330,6 +6331,118 @@ See URL `http://lesscss.org'."
+                 (const :tag "Globals of LuaJIT 2.0"
+                        ,(nth 6 flycheck-luacheck-std-globals-list))
+                 (const :tag "Intersection of globals of Lua 5.1, 5.2 and LuaJIT 2.0"
+                        ,(nth 7 flycheck-luacheck-std-globals-list))
+                 (const :tag "Union of globals of Lua 5.1, 5.2 and LuaJIT 2.0"
+                        ,(nth 8 flycheck-luacheck-std-globals-list))
+                 (const :tag "No standard globals"
+                        ,(nth 9 flycheck-luacheck-std-globals-list)))
+  :safe (lambda (value) (member value flycheck-luacheck-std-globals-list))
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-globals nil luacheck
+  "Filter out warnings related to global variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused nil luacheck
+  "Filter out warnings related to unused variables and values."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-redefined nil luacheck
+  "Filter out warnings related to redefined variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-args nil luacheck
+  "Filter out warnings related to unused arguments and loop variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-secondaries nil luacheck
+  "Filter out warnings related to unused variables set together
+with used ones."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-no-unused-globals nil luacheck
+  "Filter out warnings related to set but unused global variables."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-globals nil luacheck
+  "A list of additional globals to add on top of standard ones."
+  :type '(choice (const :tag "No additional globals" nil)
+                 (repeat :tag "Additional global"
+                         (string :tag "Global name")))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-def-option-var flycheck-luacheck-read-globals nil luacheck
+  "A list of additional globals to add on top of standard ones."
+  :type '(choice (const :tag "No additional read-only globals" nil)
+                 (repeat :tag "Additional read-only global"
+                         (string :tag "Global name")))
+  :safe #'flycheck-string-list-p
+  :package-version '(flycheck . "0.23"))
+
+(flycheck-define-checker luacheck
+  "A Lua syntax checker using luacheck.
+
+See URL `https://github.com/mpeterv/luacheck'."
+  :command ("luacheck"
+            "--no-config"
```

Comment:
This option disables the option to load an external config file. I did not introduce the option of specifying an external config file on purpose because you could set the `luacheck` output format or remove support for `codes` which would make the syntax checker fail.